src/BridgeFactory.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import {IERC1271} from "./interfaces/IERC1271.sol";
+import {IERC20} from "@forge-std/interfaces/IERC20.sol";
+import {TwoStepOwnable} from "./deployer/TwoStepOwnable.sol";
+import {SafeTransferLib} from "./vendor/SafeTransferLib.sol";
+import {MerkleProofLib} from "./vendor/MerkleProofLib.sol";
+import {Context} from "./Context.sol";
+
+contract BridgeFactory is IERC1271, TwoStepOwnable, Context {
+    using SafeTransferLib for IERC20;
+
+    address private immutable _cachedThis;
+
+    constructor() {
+        require(
+            (msg.sender == 0x4e59b44847b379578588920cA78FbF26c0B4956C &&
+                uint160(address(this)) >> 104 == 0) || block.chainid == 31337
+        );
+        _cachedThis = address(this);
+    }
+
+    modifier onlyWallet() {
+        require(address(this).code.length == 38);
```

Comment:
The snippet that you commented on over in #328 is kind of inefficient, but that's the most expedient way to detect 7702'd wallets. This modifier, on the other hand, could probably be simpler because it just has to detect proxies:
```suggestion
        require(address(this) != _cachedThis);
```

Reply:
I guess you could also name the modifier `onlyProxy` because that's a bit more specific